test/e2e/framework/network/utils.go

```diff
@@ -91,9 +91,31 @@ const (
 // NetexecImageName is the image name for agnhost.
 var NetexecImageName = imageutils.GetE2EImage(imageutils.Agnhost)
 
+// Use functional options to configure the NetworkingTest object
+type Option func(*NetworkingTestConfig)
+
+func EnableSCTP() Option {
+	return func(config *NetworkingTestConfig) {
```

Comment:
So I know you have to use a function-returning-a-function for the port ones, but I don't think it necessarily makes sense here... (especially if we end up getting rid of the port-setting options)

Reply:
We have to add another one soon `EnableDualStack()`

https://github.com/kubernetes/kubernetes/pull/95063/commits/a733598569da1974a43d4010cb92022ffc0193c3#diff-c26eb8ec2a51d898f9bbd94080b25dbef525d3ff1578bbcd0a6c5d10bc521c00R101

Reply:
was meant to simplify the review process only, each commit is not independent

Reply:
right, but I mean, for the ones that don't take arguments, you could just do

```
func EnableSCTP(config *NetworkingTestConfig) {
	config.SCTPEnabled = true
}
```

and then pass `e2enetwork.EnableSCTP` rather than `e2enetwork.EnableSCTP()`. I guess there are arguments either way.